### ISSUE TITLE
Common-lines: `n_theta` must be even.

### ISSUE THREAD

**Issue by j-c-c** [ComputationalCryoEM/ASPIRE-Python#792]
When using a common lines method `n_theta` must be even. There is a check here: https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/fad4dfd6ebf66d4983688c5ec01d99fae7167130/src/aspire/abinitio/commonline_base.py#L72-L75

But this check needs to be moved up a few line above `evaluate_t` since `n_theta // 2` is used there.

Additionally, a note should be added to the relevant docstrings.

**Comment by j-c-c** [comment]
Traceback when initializing a `CLSyncVoting` (or `CLSymmetryC3C4`) object with `n_theta` even:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[11], line 2
      1 n_theta = 359
----> 2 orient_est = CLSyncVoting(src=src, n_theta=n_theta)

File ~/Work/ASPIRE-python.n_theta_even_792/src/aspire/abinitio/commonline_sync.py:37, in CLSyncVoting.__init__(self, src, n_rad, n_theta, max_shift, shift_step)
     25 def __init__(self, src, n_rad=None, n_theta=360, max_shift=0.15, shift_step=1):
     26     """
     27     Initialize an object for estimating 3D orientations using synchronization matrix
     28 
   (...)
     35     :param shift_step: Resolution for shift estimation in pixels. Default is 1 pixel.
     36     """
---> 37     super().__init__(
     38         src,
     39         n_rad=n_rad,
     40         n_theta=n_theta,
     41         max_shift=max_shift,
     42         shift_step=shift_step,
     43     )
     44     self.syncmatrix = None

File ~/Work/ASPIRE-python.n_theta_even_792/src/aspire/abinitio/commonline_base.py:52, in CLOrient3D.__init__(self, src, n_rad, n_theta, n_check, max_shift, shift_step)
     49 self.shift_step = shift_step
     50 self.rotations = None
---> 52 self._build()

File ~/Work/ASPIRE-python.n_theta_even_792/src/aspire/abinitio/commonline_base.py:69, in CLOrient3D._build(self)
     65 # Obtain coefficients in polar Fourier basis for input 2D images
     66 self.basis = PolarBasis2D(
     67     (self.n_res, self.n_res), self.n_rad, self.n_theta, dtype=self.dtype
     68 )
---> 69 self.pf = self.basis.evaluate_t(imgs)
     70 self.pf = self.pf.reshape(self.n_img, self.n_theta, self.n_rad).T  # RCOPT
     72 if self.n_theta % 2 == 1:

File ~/Work/ASPIRE-python.n_theta_even_792/src/aspire/basis/basis.py:140, in Basis.evaluate_t(self, v)
    138 v = v.reshape(-1, *v.shape[-self.ndim :])
    139 # Compute the adjoint
--> 140 x = self._evaluate_t(v)
    141 # Restore stack shape
    142 x = x.reshape(*stack_shape, self.count)

File ~/Work/ASPIRE-python.n_theta_even_792/src/aspire/basis/polar_2d.py:120, in PolarBasis2D._evaluate_t(self, x)
    116 half_size = self.ntheta // 2
    118 pf = nufft(x, self.freqs)
--> 120 pf = pf.reshape((nimgs, self.nrad, half_size))
    121 v = np.concatenate((pf, pf.conj()), axis=1)
    123 # return v coefficients with the last dimension size of self.count

ValueError: cannot reshape array of size 2266008 into shape (789,16,179)

```

**Comment by j-c-c** [comment]
Closed by #840. See related issue #844.
